core/templates/dev/head/domain/utilities/ImageFileObjectFactory.ts

```diff
@@ -16,20 +16,29 @@
  * @fileoverview Object factory for creating image files.
  */
 
-var oppia = require('AppInit.ts').module;
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
 
-oppia.factory('ImageFileObjectFactory', [function() {
-  var ImageFile = function(filename, data) {
+export class ImageFile {
+  filename: any;
```

Comment:
This file does not have a spec file from where I could figure our, it would require backtracking and hence falls into the 'complex' category (refer [this](https://github.com/oppia/oppia/pull/7145#discussion_r304293508) thread). #7165 is for this very purpose.

Reply:
You can figure it out, and I gave you a process above where it would not take a lot of time.

Reply:
Done!